### ISSUE TITLE
Fast forward through sponsors instead of skipping them.

### ISSUE THREAD

**Issue by foxt** [ajayyy/SponsorBlock#499]
Would it be possible to fasft forward through segments (like intros or outros) instead of flat out skipping them?

**Comment by SethiPandi** [comment]
This sounds like it'd be _probably_ (not peeked at the code) be easy to do by making an option to go 2x speed for the duration using what's already built into the YouTube video player. 
Would 2x speed be fast enough though?

**Comment by theLMGN** [comment]
Not sure. setting the playbackRate starts breaking at 5, and doesn't work past 10x

**Comment by yanisallouch** [comment]
i was about to suggest something slightly different, why not just mute (+ blur the screen or something else) while still giving the ability to skip ? so we won't have to deal with playbackRate issues.

**Comment by JesFEREM** [comment]
bump

**Comment by ThiagoRangel7** [comment]
Agreed. I think we should be able to set the playback speed we want though. 2x is too slow for sponsors, as I normally watch every video in 1.75x or 2x speed. Maybe we should be able to type in a float value, just like Enhancer for YouTube extension allows.

**Comment by SameenAhnaf** [comment]
In my opinion, the video should get muted while fast forwarding the video. Another alternative could be showing a frame from every 15 or 30 seconds (just like video preview while putting the cursor on video thumbnail).

**Comment by BlackSails7** [comment]
As you can see with [this userscript](https://greasyfork.org/en/scripts/433222-improved-speed-slider-for-youtube-fix) you can easily make videos on youtube play at x4 speed. Probably up to x5... But also the speed should be modifiable in settings.

If this gets added, I guess it could be a new option here
[![speedthrough.webp](https://i.postimg.cc/kgwDTzZ8/speedthrough.webp)](https://postimg.cc/PCptx3dr)

**Comment by joshinils** [comment]
I use the addon [Video Speed Controller by codebicycle](https://addons.mozilla.org/en-US/firefox/addon/videospeed/) to speed up some otherwise boring videos.
Some channels do their editing a bit slow, I guess.

So if this does get implemented, I'd like for it to restore the previous playback rate after the segment is done.

A separate speed setting for each segment type would be great.

A different type of segment set to be played faster would also be great, instead of only skip and mute.
Then a setting for skipping speed-up-segments and speeding up skip-segments would also be beneficial I think.

**Comment by shenef** [comment]
I'm also interested in this feature.

The following are my UI/UX suggestions. They mostly overlap with the previous suggestions in this issue.

![image](https://user-images.githubusercontent.com/20268890/231524546-0a97172d-b65a-4c2e-8a6e-063432c31c82.png)

Add a "fast forward" option to the skip options and a separate dropdown (or slider) to control the playback speed.
Playback speed options should include `1.0`, `1.25`, `1.5`, `2.0` and `3.0`. 1.0 is included for those that normally watch at slower speeds and 3.0 is the max cause after that the correct playback can't be ensured (low spec devices, browser support, slow network). 

Before fast forwarding the existing playback speed must be stored so it can be restored after the segment.

While fast forwarding a skip notice should appear that offers skipping the segment using the "Skip segment" hotkey.
If a manual playback speed change is detected during fast forwarding, then the fast forward disengages and the previously stored playback speed is discarded.

---

Personally I would use fast forward mostly for fillers, self promotion and intros. I want to see those segments but watching them at normal speed (in my case x1.25) is not necessary since they are not important to me.

**Comment by CodeF53** [comment]
I don't see a point in capping speed.

I personally sometimes watch videos at 5x. Just show a warning above 3x

**Comment by joshinils** [comment]
I think it is better to have it be a multiplier of the current speed, so if i watch at a faster speed (1.3x) then it would become _even faster_ (1.3 * 1.25 = 1.625 > 1.25), not potentially slower than before.
_or_ set it as addition to current speed, i.e. 1.3 -> 1.55 -> 1.3, opposed to 1.3 -> 1.25 -> 1.3.

I dont usually watch at a higher speed, but for the cases where i am this could be annoying

**Comment by joshinils** [comment]
An example where speed up instead of skipping would be great: https://www.youtube.com/watch?v=cVUxw-KJdd4

I think a sigmoid-like curve at the start and end of such segments would work great, it slowly increases playback speed (if the segment is long enough, otherwise use a ) and at the end slowly decreases again, until the previous speed is reached again.

https://www.desmos.com/calculator/gghshe3odw
![image](https://github.com/ajayyy/SponsorBlock/assets/34302971/398f2ace-808c-4151-a69d-2080b9a03653)
![image](https://github.com/ajayyy/SponsorBlock/assets/34302971/aac5a091-9ffe-4b59-9076-2305c7428b39)

**Comment by dallytaur** [comment]
a way to implement this function

create a new menu options called fast forward

then when the function to call fast forward the jscode to make it work is as followed

`document.querySelector('video').playbackRate = <speed>;`

this can be tested in the JS console

fhen a few UI elements to have it added as said above


A small side thing is have a new skip type for slugsh rambling YT videos like Ive had number of tutroal video full of dead air and such.

for now we dont need lotta complex functions any how

**Comment by Dani3I** [comment]
It would be cool to set different speeds for different categories of market segments

**Comment by dallytaur** [comment]
https://github.com/ajayyy/SponsorBlock/issues/2262#issuecomment-3573803121_

@ajayyy Has told us that will not be added to SponserBlock to stop more people trying add this i think we close this iusse to save devs from trying add this.

**Comment by shenef** [comment]
Was there a particular reason besides the "There was a vote"?
For example that it could be an additional maintenance burden or UX issues

**Comment by dallytaur** [comment]
> Was there a particular reason besides the "There was a vote"? For example that it could be an additional maintenance burden or UX issues

when i was buggering with the code we haeva good way see a segmeant starts not good seeing ending of a segment that i saw when i tired to do it

other than that mayeb 50-80 lines of code all up to add so not a lott

but yeah vote by who the discord i guess?
